### ISSUE TITLE
Fireplace shows as a light

### ISSUE THREAD

**Issue by pateboyd** [aarons22/homebridge-bond#225]
The fireplace in bond shows as a light in Homebridge. I tried changing it in homekit, eve and a couple of others.

When I tell Siri to turn on the living room lights, the fireplace turns on as well.

**Comment by csekar15** [comment]
Were you able to solve this? I have the same issue

**Comment by dtpanic** [comment]
I have the same issue.

**Comment by mlapida** [comment]
Same issue here - can't seem to find a workaround, beyond creating a separate room for the fireplace.

**Comment by dtpanic** [comment]
The problem appears to be on line 216: https://github.com/aarons22/homebridge-bond/blob/main/src/Services.ts

**Comment by seanwalter** [comment]
I have the same issue, too

**Comment by aarons22** [comment]
There isn't a firebase service available in homekit, so a Lightbulb was chosen as the closest thing. What is expected here?

**Comment by dtpanic** [comment]
Hey Aaron!

From my perspective, a switch would be better than a lightbulb. I understand why a lightbulb was chosen over a switch because you can then use the 'brightness' to adjust flame size (if supported by the fireplace).

However for those of us without adjustable flame fireplaces, it would be preferable to have a switch. 

I actually ended up downgrading to v3.2.6, because in this version (before the flame adjustment feature was implemented) fireplaces were represented as switches. I then patched 3.2.6 with the bug fix from 3.2.10.

**Comment by seanwalter** [comment]
Exposing it as a (plain) switch would solve this issue.

That said, the best option would most likely be to have the choice to let the user specify it as a light, a fan or a switch, in accordance with their preference. I have seen this option in the settings for other HomeKit accessories, including native integrations as well as other accessories exposed by Homebridge.

**Comment by aarons22** [comment]
@dtpanic makes sense! I ended up landing there after thinking about it a bit further. Would you be open to testing a beta release if I made the change? I don't have a fireplace control myself, so it's tricky to test.

**Comment by aarons22** [comment]
@seanwalter as mentioned above, I think the easiest approach is to implement as a Lightbulb when a flame control is present, otherwise, implement it as Switch. would that work for you? Custom configs per accessory type get more complicated and I'd prefer a simple fix here

**Comment by dtpanic** [comment]
@aarons22 I would be happy to test the beta! 

I agree with your proposed solution - if flame control is available, present the fireplace as a lightbulb. If not, present it as a switch.

**Comment by seanwalter** [comment]
Thank you, @aarons22 and @dtpanic!

My fireplace does have an adjustable flame so my proposed solution wouldn’t actually work for me :)

If you implement the switch and the dimmer I can probably hack what I need.

Alternatively, while visually weird, if instead of a dimmer (light) you used a fan (climate) I think you’d get the exact functionality and could specify the number of flame adjustments. But if that’s hard or just too weird, the switch and dimmer will work for me.

Thank you!!

ps - happy to beta test

**Comment by aarons22** [comment]
@dtpanic @seanwalter beta is published! https://github.com/aarons22/homebridge-bond/releases/tag/3.2.11-beta.2

**Comment by dtpanic** [comment]
That was quick! I should have mentioned I'm not currently at my home with the Bond device, so I can't test it end-to-end. However, I did install the beta and confirmed that:

1. The fireplace was recognized and added as a switch.
2. I'm able to turn on and off the switch in the Home app.
3. There are no errors in the homebridge logs.

The only thing I can't check from here is whether the on/off commands are being successfully sent to the Bond device. I will check that next time I visit that location (could be a few weeks from now). 

If there's anything else I can do to test remotely, let me know. Thanks for all your work on this!

**Comment by csekar15** [comment]
I’m at home and will install and test.

**Comment by csekar15** [comment]
It seems to be working fine. If I say “turn on all lights”, it does NOT turn on the fireplace like it used to. If I say “turn on fireplace”, it turns on just the fireplace. Turning OFF works the same way as well.

Thanks for the speedy fix.

I do see two controls for fireplace now. One looks like a switch and one looks like a light. Is that supposed to be like that? 

Chandra

**Comment by seanwalter** [comment]
I installed and tested, and can verify the dimmer is still passed through and works. I’ll need a few days to set up my Bond to try out an alternative non-dimmer configuration, but if others say the switch is appropriately passed through, it’s probably fine — I don’t think my future setup will break the new functionality.
